platforms/chibios/drivers/eeprom/eeprom_stm32_L0_L1.c

```diff
@@ -86,17 +86,37 @@ void eeprom_read_block(void *buf, const void *addr, size_t len) {
 }
 
 void eeprom_write_block(const void *buf, void *addr, size_t len) {
-    STM32_L0_L1_EEPROM_Unlock();
+    // use word-aligned write to overcome issues with writing null bytes
+    uint32_t start_addr = (uint32_t)addr;
+    uint32_t end_addr   = start_addr + len;
 
-    for (size_t offset = 0; offset < len; ++offset) {
-        // Drop out if we've hit the limit of the EEPROM
-        if ((((uint32_t)addr) + offset) >= STM32_ONBOARD_EEPROM_SIZE) {
-            break;
-        }
+    uint32_t aligned_start = start_addr & ~0x3;
+    uint32_t aligned_end   = (end_addr + 3) & ~0x3;
 
-        STM32_L0_L1_EEPROM_WaitNotBusy();
-        EEPROM_BYTE(addr, offset) = BUFFER_BYTE(buf, offset);
+    // Don't write if we've hit the limit of the EEPROM
+    if (aligned_end >= STM32_ONBOARD_EEPROM_SIZE) {
+        return;
     }
```

Comment:
This changes the behavior for partially out-of-range writes (the old code still wrote the part which was in range, the new code drops the whole write).  However, this probably does not matter (and some other implementations, like `wear_leveling`, also drop whole writes that way).

Reply:
After thinking about it, I don't think we need this guard anyway, since the EEPROM's end will be aligned to the end of the last word. And, even if there is a region of EEPROM being used for something else past STM32_ONBOARD_EEPROM_SIZE, we won't overwrite it with this code anyway.

Reply:
OK, I did have to add back something like the old EEPROM limit check, please review again.